controllers/mysqlcluster_controller.go

```diff
@@ -251,13 +252,8 @@ func (r *MySQLClusterReconciler) reconcileV1(ctx context.Context, req ctrl.Reque
 		return ctrl.Result{}, err
 	}
 
-	if cluster.Status.ReconcileInfo.Generation != cluster.Generation {
-		cluster.Status.ReconcileInfo.Generation = cluster.Generation
-		cluster.Status.ReconcileInfo.ReconcileVersion = 1
-		if err := r.Status().Update(ctx, cluster); err != nil {
-			log.Error(err, "failed to update reconciliation info")
-			return ctrl.Result{}, err
-		}
+	if err := r.UpdateStatus(ctx, req, cluster); err != nil {
```

Comment:
In the following case, the MOCO will not update the StatefulSet's status synchronously, 

When after updating a MySQL cluster resource:
- `reconcileV1StatefulSet()` succeeds, and a StatefulSet is updated.
- But `reconcileV1BackupJob()` fails (e.g., Invalid BackupPolicy is specified.)

Reply:
@masa213f 
Should I update Status after `reconcileV1StatefulSet()`?

By the way, in original code, `Status.ReconcileInfo.Generation` wil not be updated when same situation.
Isn't it no problem?

Reply:
> Should I update Status after reconcileV1StatefulSet()?

It seems to work fine.

> By the way, in original code, Status.ReconcileInfo.Generation wil not be updated when same situation.
Isn't it a problem?

We should not update `Status.ReconcileInfo.Generation` until a reconciliation succeeds.
If we update the generation early, the Argo CD health check will incorrectly pass, despite reconciliation errors occurring.

Reply:
I understand.
I moved `updateStatus()` just after reconcileV1StatefulSet(), and I moved codes that updates `Status.ReconcileInfo.Generation` in original location.